src/java_bytecode/java_types.h

```diff
@@ -30,6 +30,48 @@ class java_class_typet:public class_typet
   {
     return set(ID_access, access);
   }
+
+  class java_lambda_method_handlet : public exprt
```

Comment:
Noting my reservations about inheriting from `exprt` here (my preference would be `irept` since it isn't a node in an AST but we still need irep functionality to store it in a `class_typet`). However interested in seeing other perspectives on this

Reply:
After an offline discussion, a reasonable solution seems to be holding the method handle as a `symbol_exprt` (which only has an identifier of the referred method).